engine/transaction/draft/create_paymail_draft_test.go

```diff
@@ -0,0 +1,264 @@
+
+	t.Run("return draft with payment to valid paymail address", func(t *testing.T) {
+		// given:
+		paymailHostResponse := paymailmock.P2PDestinationsForSats(transactionSatoshiValue)
+
+		paymailClient := paymailmock.CreatePaymailClientService(testDomain)
+		paymailClient.WillRespondWithP2PCapabilities()
+		paymailClient.
+			WillRespondOnCapability(paymail.BRFCP2PPaymentDestination).
+			With(paymailHostResponse)
+
+		// and:
+		draftService := draft.NewDraftService(paymailClient, tester.Logger())
+
+		// and:
+		spec := &draft.TransactionSpec{
+			Outputs: outputs.NewSpecifications(&outputs.Paymail{
+				To:       recipient,
+				Satoshis: transactionSatoshiValue,
+			}),
+		}
+
+		// when:
+		draftTx, err := draftService.Create(context.Background(), spec)
+
+		// then:
+		require.NoError(t, err)
+		require.NotNil(t, draftTx)
+
+		// and:
+		annotations := draftTx.Annotations
+		require.Len(t, annotations.Outputs, 1)
+
+		// and:
+		annotation := annotations.Outputs[0]
+		require.Equal(t, transaction.BucketBSV, annotation.Bucket)
+		require.NotNil(t, annotation.Paymail)
+		assert.Equal(t, recipient, annotation.Paymail.Receiver)
+		assert.Equal(t, paymailHostResponse.Reference, annotation.Paymail.Reference)
+
+		// debug:
+		t.Logf("BEEF: %s", draftTx.BEEF)
+
+		// when:
+		tx, err := sdk.NewTransactionFromBEEFHex(draftTx.BEEF)
+
+		// then:
+		require.NoError(t, err)
+		require.Len(t, tx.Outputs, 1)
+
+		// and:
+		output := tx.Outputs[0]
+		require.EqualValues(t, transactionSatoshiValue, output.Satoshis)
+		require.Equal(t, paymailHostResponse.Scripts[0], output.LockingScriptHex())
+	})
+
+	t.Run("return draft with payment with multiple outputs from valid paymail address", func(t *testing.T) {
+		const firstOutputSatoshiValue = uint(1)
+		const secondOutputSatoshiValue = uint(2)
+		const paymentSatoshiValue = firstOutputSatoshiValue + secondOutputSatoshiValue
+
+		// given:
+		paymailHostResponse := paymailmock.P2PDestinationsForSats(firstOutputSatoshiValue, secondOutputSatoshiValue)
+
+		paymailClient := paymailmock.CreatePaymailClientService(testDomain)
+		paymailClient.WillRespondWithP2PCapabilities()
+		paymailClient.
+			WillRespondOnCapability(paymail.BRFCP2PPaymentDestination).
+			With(paymailHostResponse)
+
+		// and:
+		draftService := draft.NewDraftService(paymailClient, tester.Logger())
+
+		// and:
+		spec := &draft.TransactionSpec{
+			Outputs: outputs.NewSpecifications(&outputs.Paymail{
+				To:       recipient,
+				Satoshis: paymentSatoshiValue,
+			}),
+		}
+
+		// when:
+		draftTx, err := draftService.Create(context.Background(), spec)
+
+		// then:
+		require.NoError(t, err)
+		require.NotNil(t, draftTx)
+
+		// and:
+		annotations := draftTx.Annotations
+		require.Len(t, annotations.Outputs, 2)
+		assert.Equal(t, transaction.BucketBSV, annotations.Outputs[0].Bucket)
+		assert.Equal(t, transaction.BucketBSV, annotations.Outputs[1].Bucket)
+		// TODO: add assertions for paymail annotations
```

Comment:
Is it a leftover or you're gonna do it in te next task?

Reply:
leftover I don't know why I read it previously as add assertions for sender 🙈

Reply:
actually I forgot to add those assertions, now they are added.